File: python/1d-dynamic-programming/house-robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        firstcost, secondcost = 0, 0
        for i in nums:
            currentdecision = max(firstcost + i, secondcost)
            firstcost = secondcost
            secondcost = currentdecision

        return currentdecision

File: c++/1d-dynamic-programming/house-robber.cc
class Solution {
public:
    int rob(vector<int>& nums) {
        std::vector<int> house(nums.size() + 2, 0);
        for (int i = 2; i < house.size(); ++i) {
            house[i] = std::max(house[i-2] + nums[i-2], house[i-1]);
        }
        return house[house.size() - 1];
    }
};
